### PR TITLE
docs: add templates and update package information

### PR DESCRIPTION
**Pull request by RasmusSkytte**
This PR does a few things.

1) It adds the documentation templates for the package
2) It updates the package description with author information and more information about the package
3) It updates the readme to describe the package

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+    checkmate,
+    glue,
+    purrr,
+    stringr
 Suggests: 
     knitr,
     rmarkdown,
     spelling
 VignetteBuilder: knitr
-URL: https://github.com/RasmusSkytte/diseasy,
```
Comment (marcusmunch) [inline DESCRIPTION:18]: Is there a reason the URL field was removed?

File: DESCRIPTION [hunk 2]
```diff
@@ -1,23 +1,27 @@
 Package: diseasy
-Title: What the Package Does (One Line, Title Case)
+Title: disease ensemble modelling made easy
```
Comment (marcusmunch) [inline DESCRIPTION:2]: Title case:
```suggestion
Title: Disease Ensemble Modelling Made Easy
```

File: DESCRIPTION (hunk 1 above)
Comment (RasmusSkytte) [reply DESCRIPTION:18]: It is no longer the valid URL.
Proper URLs for all fields will be added as part of the release issue (when created)

Comment (RasmusSkytte) [reply DESCRIPTION:18]: Release issue is created now, and the URL is included again

**Pull request merged**